### PR TITLE
[Reader IA] Update selected content stream based on menu item

### LINKED ISSUE #19605: [Reader IA][Drop-down menu] Replace ViewPager with drop-down menu logic
pcdRpT-4tr-p2

Depends on https://github.com/wordpress-mobile/WordPress-Android/issues/19604

image

### PR DESCRIPTION
**Pull request by RenanLukas**
Part of #19605 

> [!CAUTION]
> ~Should not be merged before https://github.com/wordpress-mobile/WordPress-Android/pull/19805~

`ReaderViewModel` test will be updated in my next PR.

https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/888fc7df-29f0-4e62-a945-c664699fccdf

-----

## To Test:
1 - Open reader
2 - Select menu items in multiple states (e.g. "Saved" with and without saved posts): the posts list should be updated correctly
3 - Make sure you have no saved posts and open "Saved" menu item
4 - Tap on "Go to subscriptions": it should open "Subscriptions" menu item correctly

-----

## Regression Notes

1. Potential unintended areas of impact
- Reader existing fragments

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing

3. What automated tests I added (or what prevented me from doing so)
- Unit tests will be implemented in another PR 

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by thomashorta — COMMENTED** [review]
I reviewed the code and I think this is a good first step in hooking the feed logic with the new UI although I hope we can improve some things in the future, such as the usage of `ReaderTag` for representing the menu items and the usage of the term "tab" in some places, even though we moved from tabs to menu items.

I noticed the CI was not passing so I pushed some changes to address detekt issues.

Besides that, I found out while testing that the pull-to-refresh causes the feed to reset to "Discover" and when switching between bottom tabs (`My Site`, `Reader`, etc), the state of the `Reader` is not saved/restored, which I am not sure is expected. cc @develric so we decide which of those issues we will tackle in this PR and which we will open tickets and tackle later (still during development/before reaching trunk).

**Review by thomashorta — APPROVED** [review]
I am approving this PR, since I talked with @develric and we decided that the bugs I found (last comment) should not block this PR and I created GH issues which should be fixed in the feature branch before merging to `trunk`:
- https://github.com/wordpress-mobile/WordPress-Android/issues/19852
- https://github.com/wordpress-mobile/WordPress-Android/issues/19851

**Comment by RenanLukas** [comment]
Thanks for the review and for extracting the menu items creation to a separate class, @thomashorta 

I've reviewed your commits and code LGTM. I also went through the testing steps again and it's working as expected.

**Pull request merged**
